cmd/config_file.go

```diff
@@ -0,0 +1,173 @@
+	viper.SetConfigName("cn")         // name of config file (without extension)
+	viper.AddConfigPath("/etc/cn/")   // path to look for the config file in
+	viper.AddConfigPath("$HOME/.cn/") // call multiple times to add many search paths
+	viper.AddConfigPath(".")          // optionally look for config in the working directory
+	// Let's try to read a configuration file
+	if viper.ReadInConfig() == nil {
+		// We return the configuration file found
+		return viper.ConfigFileUsed()
+	}
+
+	// No configuration file is used aka compatiblity mode
+	return ""
+}
+
+// Set the default values for defined types
+// If the configuration file is missing, this section will generated the mandatory elements
+func setDefaultConfig() {
+	viper.SetDefault(FLAVORS+".default.use_default", "true") // All containers inherit from default
+	viper.SetDefault(FLAVORS+".default.memory_size", "512MB")
+	viper.SetDefault(FLAVORS+".default.cpu_count", 1)
+	viper.SetDefault(IMAGES+".default.imageName", DEFAULTIMAGE)
+}
+
+func getStringFromConfig(group string, item string, name string) string {
+	var value = ""
+	// If we are requested to get the status of use_default, we cannot call useDefault ;)
+	if name == "use_default" || useDefault(group, item) {
+		value = viper.GetString(group + ".default." + name)
+	}
+	itemValue := viper.GetString(group + "." + item + "." + name)
+	if len(itemValue) > 0 {
+		value = itemValue
+	}
+	return value
+}
+
+func getInt64FromConfig(group string, item string, name string) int64 {
+	var value int64
+	var foundValue = false
+	if useDefault(group, item) {
+		// We need to ensure the key exists unless that could populate a 0 value
+		if viper.Get(group+".default."+name) != nil {
+			value = viper.GetInt64(group + ".default." + name)
+			foundValue = true
+		}
+	}
+	// We need to ensure the key exists unless that could populate a 0 value
+	if viper.Get(group+"."+item+"."+name) != nil {
+		value = viper.GetInt64(group + "." + item + "." + name)
+		foundValue = true
+	}
+
+	if !foundValue {
+		panic(name + " int64 value in " + item + "doesn't exists")
+	}
+	return value
+}
+
+func useDefault(group string, item string) bool {
+	useDefaultValue := getStringFromConfig(group, item, "use_default")
+	if (len(useDefaultValue) > 0) && (useDefaultValue == "true") {
+		return true
+	}
+	return false
+}
+
+func getStringMapFromConfig(group string, item string, name string) map[string]interface{} {
+	var defaultConfig = make(map[string]interface{})
+	if useDefault(group, item) {
+		defaultConfig = viper.GetStringMap(group + ".default" + "." + name)
+	}
+	itemValues := viper.GetStringMap(group + "." + item + "." + name)
+	if len(itemValues) > 0 {
+		for key, value := range itemValues {
+			defaultConfig[key] = value
+		}
+	}
+	return defaultConfig
+}
+
+func getMemorySize(containerFlavor string) string {
+	return getStringFromConfig(FLAVORS, containerFlavor, "memory_size")
+}
+
+func getMemorySizeInBytes(containerFlavor string) int64 {
```

Comment:
Shouldn't we put this in `utils.go` instead?

Reply:
done.